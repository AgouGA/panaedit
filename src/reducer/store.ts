export type Scene = {
  title: string;
  panorama: string;
  northOffset?: number;
  hotspots?: any[] | null;
};

export type Store = {
  default: {
    firstScene: string;
    sceneFadeDuration: number;
    type: "equirectangular";
    autoLoad: boolean;
    compass: boolean;
    hotSpotDebug: boolean;
    hfov: number;
    vfow: number;
    minPitch: number;
    maxPitch: number;
    basePath: string;
    imagePath: string;
  };
  scenes: { [key: string]: Scene };
  articles: object[];
  editor: {
    currentScene: string;
    workingDirectory: null | any;
  };
};

export type Action = {
  type: string;
  payload: unknown;
};

const storeDefaults: Store = {
  default: {
    firstScene: "demo",
    sceneFadeDuration: 1000,
    type: "equirectangular",
    autoLoad: true,
    compass: true,
    hotSpotDebug: true,
    hfov: 110,
    vfow: 100,
    minPitch: -98,
    maxPitch: 98,
    basePath: "assets/panorama/",
    imagePath: "assets/images/",
  },
  scenes: {
    demo: {},
  },
  articles: [],
  editor: {
    currentScene: "",
    workingDirectory: null,
  },
};

export default storeDefaults;
