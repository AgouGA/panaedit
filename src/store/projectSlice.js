import { createReducer, createSlice } from '@reduxjs/toolkit'

export const projectSlice = createSlice({
	name: 'project',
	initialState: {},
	reducers: {
		test: (state) => state
	}
})

export default projectSlice.reducer

export const { test } = projectSlice.actions
