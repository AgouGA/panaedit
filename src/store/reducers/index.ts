/*
 *
 */

export * from './article'
export * from './editor'
export * from './hotspot'
// index.ts
export * from './photo'
export * from './project'
export * from './scene'
