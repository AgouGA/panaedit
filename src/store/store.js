import projectReducer from './reducer'

export default configureStore({
	reducer: {
		project: projectReducer
	}
})
