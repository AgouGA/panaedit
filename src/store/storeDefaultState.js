const storeDefaultState = {
	default: {
		firstScene: '',
		sceneFadeDuration: 1000,
		type: 'equirectangular',
		autoLoad: true,
		compass: true,
		hotSpotDebug: true,
		hfov: 110,
		vfow: 100,
		minPitch: -98,
		maxPitch: 98,
		basePath: 'assets/panorama/',
		imagePath: 'assets/images/'
	},
	scenes: {},
	articles: [],
	editor: {
		currentScene: ''
	}
}

export default storeDefaults
