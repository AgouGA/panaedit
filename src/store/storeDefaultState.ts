const storeDefaultState: State = {
	default: {
		firstScene: '',
		sceneFadeDuration: 1000,
		type: 'equirectangular',
		autoLoad: true,
		compass: true,
		hotSpotDebug: true,
		hfov: 110,
		vfov: 100,
		minPitch: -98,
		maxPitch: 98,
		basePath: '/panoramas'
	},
	scenes: {},
	articles: [],
	editor: {
		activeScene: ''
	}
}

export default storeDefaultState
