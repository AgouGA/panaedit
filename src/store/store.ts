import { legacy_createStore } from "redux";
import { devToolsEnhancer } from "redux-devtools-extension";
import { reducer } from "./reducer";

export type HotSpotType = {
    y: number;
    x: number;
};

export type SceneType = {
    title: string;
    panorama: string;
    northOffset?: number;
    hotspots: HotSpotType[] | null;
};

export type Store = {
    default: {
        firstScene: string;
        sceneFadeDuration: number;
        type: "equirectangular";
        autoLoad: boolean;
        compass: boolean;
        hotSpotDebug: boolean;
        hfov: number;
        vfow: number;
        minPitch: number;
        maxPitch: number;
        basePath: string;
        imagePath: string;
    };
    scenes: { [key: string]: SceneType } | null;
    articles: object[] | null;
    editor: {
        currentScene: string;
        workingDirectory: null | any;
    };
};

export const storeDefaults: Store = {
    default: {
        firstScene: "",
        sceneFadeDuration: 1000,
        type: "equirectangular",
        autoLoad: true,
        compass: true,
        hotSpotDebug: true,
        hfov: 110,
        vfow: 100,
        minPitch: -98,
        maxPitch: 98,
        basePath: "assets/panorama/",
        imagePath: "assets/images/",
    },
    scenes: null,
    articles: null,
    editor: {
        currentScene: "",
        workingDirectory: null,
    },
};

export const store = legacy_createStore(reducer, devToolsEnhancer({}));
