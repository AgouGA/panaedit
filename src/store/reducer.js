import { createSlice } from '@reduxjs/toolkit'
import storeDefaultState from './storeDefaultState'

export const projectSlice = createSlice({
	name: 'project',
	initialState: storeDefaultState,
	reducers: {
		reset: (state) => storeDefaultState
	}
})

export default projectSlice.reducer

export const { reset } = projectSlice.actions
