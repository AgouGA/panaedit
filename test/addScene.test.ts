import { _reset, _addScene } from '../src/store/reducers'
import { describe, it } from 'vitest'
import storeDefaultState from '../src/store/storeDefaultState'

describe('test reducer functions', () => {
	it('_reset should return the default project', async ({ expect }) => {
		const output = _reset({} as any)

		expect(output).toMatchSnapshot()
	})

	it('add scene should add new key to scenes obj', async ({ expect }) => {
		const scene: Scene = {
			hotSpots: [],
			panorama: './assets/testimage.jpg',
			id: 'testid',
			name: 'test object',
			type: 'link'
		}

		const project = storeDefaultState

		const output = _addScene(project, { payload: { scene: scene } })

		expect(output).toMatchSnapshot()
	})
})
