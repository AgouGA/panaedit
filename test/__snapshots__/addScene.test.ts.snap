// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test reducer functions > _reset should return the default project 1`] = `
{
  "articles": [],
  "default": {
    "autoLoad": true,
    "basePath": "/panoramas",
    "compass": true,
    "firstScene": "",
    "hfov": 110,
    "hotSpotDebug": true,
    "maxPitch": 98,
    "minPitch": -98,
    "sceneFadeDuration": 1000,
    "type": "equirectangular",
    "vfov": 100,
  },
  "editor": {
    "activeScene": "",
  },
  "scenes": {},
}
`;

exports[`test reducer functions > add scene should add new key to scenes obj 1`] = `
{
  "articles": [],
  "default": {
    "autoLoad": true,
    "basePath": "/panoramas",
    "compass": true,
    "firstScene": "",
    "hfov": 110,
    "hotSpotDebug": true,
    "maxPitch": 98,
    "minPitch": -98,
    "sceneFadeDuration": 1000,
    "type": "equirectangular",
    "vfov": 100,
  },
  "editor": {
    "activeScene": "",
  },
  "scenes": {
    "testimage": {
      "hotSpots": [],
      "id": "testid",
      "name": "test object",
      "northOffset": 0,
      "panorama": "./assets/testimage.jpg",
      "type": "link",
    },
  },
}
`;
